src/library_pthread.js

```diff
@@ -238,11 +238,13 @@ var LibraryPThread = {
       // module loaded.
       PThread.tlsInitFunctions.forEach((f) => f());
     },
-    // Loads the WebAssembly module into the given list of Workers.
+    // Loads the WebAssembly module into the given Worker.
     // onFinishedLoading: A callback function that will be called once all of
     //                    the workers have been initialized and are
     //                    ready to host pthreads.
-    loadWasmModuleToWorker: function(worker, onFinishedLoading) {
+    loadWasmModuleToWorker: function(worker) {
+      let onFinishedLoading;
+      let ready = new Promise(resolve => onFinishedLoading = resolve);
```

Comment:
Can we just `return new Promise(resolve => onFinishedLoading = resolve);` below?

Reply:
Sure, we could, it just felt like it might be confusing to read because it then relies on the fact that `worker.onmessage` executes in the next event loop so the `onFinishedLoading` variable gets a chance to be assigned before that.

Reply:
The even cleaner option would be to wrap this entire function body into `new Promise(onFinishedLoading => { ... })` but I'm not sure if indentation is worth it.